### PR TITLE
Backport Summer Work (Foxy/Rolling) to Noetic

### PR DESCRIPTION
**Pull request by AdamPettinger**
Just changes for compiling and tests passing in Noetic.

Maybe interesting to @EAkita or @PaperFanz

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,47 +1,65 @@
...
 
-find_package(ament_cmake REQUIRED)
-ament_package_xml()
-
+add_compile_options(-std=c++14)
 if(NOT "${CMAKE_CXX_STANDARD}")
-  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD 14)
 endif()
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```
Comment (PaperFanz) [inline CMakeLists.txt]: Are both of these lines necessary? I thought CMake perferred set(standard) over add_compile_options due to portability

File: include/affordance_primitives/affordance_utils.hpp [hunk 2]
```diff
@@ -35,8 +35,8 @@
 
 #include <optional>
 
-#include <geometry_msgs/msg/pose_stamped.hpp>
-#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/TwistStamped.h>
```
Comment (PaperFanz) [inline include/affordance_primitives/affordance_utils.hpp:39]: don't you just love when namespaces change on you 🙃

File: include/affordance_primitives/screw_conversions.hpp [hunk 3]
```diff
@@ -58,7 +58,7 @@ inline Eigen::Matrix3d getSkewSymmetricMatrix(const Eigen::Vector3d& vec)
   return output;
 }
 
-inline std::string twistToStr(const geometry_msgs::msg::TwistStamped& twist)
+inline std::string twistToStr(const geometry_msgs::TwistStamped& twist)
```
Comment (PaperFanz) [inline include/affordance_primitives/screw_conversions.hpp:61]: when using inline functions it's best to add the `-Winline` compiler flag so you know when your inline directives are being ignored. most probably not in this case but a good practice in larger codebases

**Review by PaperFanz — APPROVED** [review]
Great stuff, liking the gtest additions as well!

File: CMakeLists.txt (hunk 1 above)
Comment (AdamPettinger) [reply CMakeLists.txt]: Good catch, totally slipped through on copy-paste-modify error

**Review by PaperFanz — APPROVED** [review]
LGTM!

**Pull request merged**